test/utils/Context.test.js

```diff
@@ -1,15 +1,12 @@
 const { ethers } = require('hardhat');
 const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
 
-const { shouldBehaveLikeRegularContext } = require('../utils/Context.behavior');
+const { shouldBehaveLikeRegularContext } = require('./Context.behavior');
 
 async function fixture() {
   const [sender] = await ethers.getSigners();
-
   const context = await ethers.deployContract('ContextMock', []);
-  const caller = await ethers.deployContract('ContextMockCaller', []);
```

Comment:
caller is only used by the `shouldBehaveLikeRegularContext`, so I propose we deploy it there directly.